src/app/MessageDef/Parser.h

```diff
@@ -73,6 +74,23 @@ class Parser
         return GetSimpleValue(aContextTag, chip::TLV::kTLVType_UnsignedInteger, apLValue);
     };
 
+    template <typename T>
+    CHIP_ERROR GetUnsignedInteger(const uint8_t aContextTag, Optional<T> & aValue) const
+    {
+        T value;
+        CHIP_ERROR err = GetSimpleValue(aContextTag, chip::TLV::kTLVType_UnsignedInteger, &value);
+        aValue.ClearValue();
+        if (err == CHIP_NO_ERROR)
+        {
+            aValue.SetValue(value);
+        }
+        if (err != CHIP_ERROR_END_OF_TLV)
+        {
+            return err;
+        }
+        return CHIP_NO_ERROR;
```

Comment:
SetValue does a clear anyway, so this might be better as:
```suggestion
        if (err == CHIP_NO_ERROR)
        {
            aValue.SetValue(value);
        }
        else if (err == CHIP_ERROR_END_OF_TLV)
        {
            aValue.ClearValue();
        }
        else
        {
            return err;
        }
        return CHIP_NO_ERROR;
```
or so.